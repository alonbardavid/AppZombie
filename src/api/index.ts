export * from './login';
export * from './users';
export * from './posts';

import * as Schema from './endpoint-schema';
export {Schema};