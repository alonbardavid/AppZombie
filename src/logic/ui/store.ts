import React from 'react';
import {observable} from "mobx";


export class UiStore {
    @observable
    navState:any;

}