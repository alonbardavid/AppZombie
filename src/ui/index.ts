export * from './form';
export * from './layout';
export * from './text';
export * from './buttons';