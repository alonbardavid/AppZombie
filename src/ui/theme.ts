
export const colors = {
    primary: "#0092FF",
    error: "#FF0000",
    text: "#000000",
    textReverse:"#FFFFFF",
    background: "#cbcbcb",
    button:{
        disabled: "#6b6b6b",
        active:"#0092FF"
    }
};

export const sizes = {
    text: 20,
    textSmall: 15,
    pageTitle:30,
    layoutPadding:20,
    inputPadding:5
};