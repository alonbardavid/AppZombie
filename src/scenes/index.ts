import {registerScreen} from 'intent-router/src/intent-router';
import {HomeScene} from './home';
import {LoginScene} from "./login";
import {PostsScene} from "./posts";

registerScreen(HomeScene);
registerScreen(LoginScene);
registerScreen(PostsScene);
export {HomeScene,LoginScene, PostsScene};