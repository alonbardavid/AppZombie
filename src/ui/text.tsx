import React from 'react';
import {Text, StyleSheet, TextProperties} from 'react-native';
import {sizes,colors} from 'src/ui/theme';



export class PageTitle extends React.PureComponent<TextProperties> {

    render(){
        const {style,children,...rest} = this.props;

        return <Text style={[styles.pageTitle,style]} {...rest} >
            {children}
        </Text>
    }
}
export interface TextRegularProperties extends TextProperties {
    small?:boolean
}
export class TextRegular extends React.PureComponent<TextRegularProperties> {

    render(){
        const {style,small,children,...rest} = this.props;
        return <Text style={[styles.regular,{
            fontSize:small?sizes.textSmall: sizes.text
        },style]} {...rest} >
            {children}
        </Text>
    }
}
export class TextError extends React.PureComponent<TextProperties>{
    render(){
        const {style,children,...rest} = this.props;
        return <Text style={[styles.error,style]} {...rest} >
            {children}
        </Text>
    }
}
const styles = StyleSheet.create({
    pageTitle:{
        color:colors.text,
        fontSize:sizes.pageTitle,
        fontWeight:"bold"
    },
    regular:{
        color:colors.text,
        fontSize:sizes.text
    },
    error:{
        color:colors.error,
        fontSize:sizes.textSmall
    }
});