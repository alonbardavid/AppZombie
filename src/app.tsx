//import './logic/routes';
//import {startApp} from 'intent-router/module';
/*try {
    startApp().catch(e => console.error(e));
} catch(e){
    console.error(e);
}*/
import {Navigation} from 'react-native-navigation';
import {RootScene} from 'src/scenes/root';

console.log("hey")
Navigation.registerComponent('AppZombie', () => RootScene);
Navigation.startSingleScreenApp({
    screen:{
        screen:"AppZombie",
        navigatorStyle:{
            navBarHidden:true
        }
    }
})