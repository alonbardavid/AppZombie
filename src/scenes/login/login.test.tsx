import React from 'react';
import {CTAButton} from "src/ui";
import renderer from "react-test-renderer";
import {  shallow } from 'enzyme';
import {LoginScene} from './';

test('initial scene',()=>{
    const tree = renderer.create(
        <LoginScene submit={async ()=>{}}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
test('scene with error',()=>{
    const wrapper = shallow(
        <LoginScene submit={async ()=>{throw new Error("custom error")}}/>
    );
    wrapper.setProps({filled:true});
    expect(wrapper).toMatchSnapshot();
});
test('clicking on login button when full should trigger submit',async ()=>{
    const wrapper = shallow(
        <LoginScene submit={async ()=>{throw new Error("custom error")}}/>
    );
    await wrapper.find(CTAButton).props().onPress();
    expect(wrapper).toMatchSnapshot();
});

