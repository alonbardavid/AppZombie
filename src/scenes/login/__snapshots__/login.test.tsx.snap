// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking on login button when full should trigger submit 1`] = `
<FormLayout>
  <PageTitle>
    My App!
  </PageTitle>
  <TextRegular>
    Please enter your credentials to login.
  </TextRegular>
  <TextRegular
    small={true}
  >
    (hint: guest/guest)
  </TextRegular>
  <StandardInput
    editable={true}
    label="Username"
    onChangeText={[Function]}
    value=""
  />
  <StandardInput
    editable={true}
    label="Password"
    onChangeText={[Function]}
    secureTextEntry={true}
    value=""
  />
  <CTAButton
    loading={false}
    onPress={[Function]}
  >
    Login
  </CTAButton>
</FormLayout>
`;

exports[`initial scene 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "justifyContent": "center",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#cbcbcb",
        "flex": 1,
        "height": "100%",
        "padding": 20,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 30,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      My App!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          Object {
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Please enter your credentials to login.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          Object {
            "fontSize": 15,
          },
          undefined,
        ]
      }
    >
      (hint: guest/guest)
    </Text>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
            },
            Object {},
          ]
        }
      >
        Username
      </Text>
      <TextInput
        editable={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
            },
            Object {},
          ]
        }
      >
        Password
      </Text>
      <TextInput
        editable={true}
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0092FF",
          "opacity": 1,
          "padding": 5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              Object {
                "fontSize": 20,
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
              ],
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`scene with error 1`] = `
<FormLayout>
  <PageTitle>
    My App!
  </PageTitle>
  <TextRegular>
    Please enter your credentials to login.
  </TextRegular>
  <TextRegular
    small={true}
  >
    (hint: guest/guest)
  </TextRegular>
  <StandardInput
    editable={true}
    label="Username"
    onChangeText={[Function]}
    value=""
  />
  <StandardInput
    editable={true}
    label="Password"
    onChangeText={[Function]}
    secureTextEntry={true}
    value=""
  />
  <CTAButton
    loading={false}
    onPress={[Function]}
  >
    Login
  </CTAButton>
</FormLayout>
`;
